scripts/irods/test/test_native_authentication.py

```diff
@@ -354,3 +354,51 @@ def test_password_extend_lifetime_set_to_false_invalidates_other_authentications
 
             # Re-authenticate as the session user to make sure things can be cleaned up.
             self.auth_session.assert_icommand(['iinit'], 'STDOUT', 'iRODS password', input=f'{self.auth_session.password}\n')
+
+    def test_password_max_time_can_exceed_1209600__issue_3742_5096(self):
+        # Note: This does NOT test the TTL as this would require waiting for the password to expire (2 weeks + 1 hour).
+        # The test is meant to ensure that a TTL greater than 1209600 is allowed with iinit when it is so configured.
+
+        max_time_option_name = 'password_max_time'
+
+        # Stash away the original configuration for later...
+        original_max_time = self.admin.assert_icommand(
+            ['iadmin', 'get_grid_configuration', self.configuration_namespace, max_time_option_name], 'STDOUT')[1].strip()
+
+        try:
+            # The test value is 2 hours more than the default in order to try a TTL value 1 greater and 1 less than the
+            # configured password_max_time while still remaining above 1209600 to show that there is nothing special
+            # about that value.
+            base_ttl_in_hours = 336 + 2
+            base_ttl_in_seconds = base_ttl_in_hours * 3600
+
+            # Set password_max_time to the value for the test.
+            option_value = str(base_ttl_in_seconds)
+            self.admin.assert_icommand(
+                ['iadmin', 'set_grid_configuration', self.configuration_namespace, max_time_option_name, option_value])
+
+            # Note: The min/max check does not occur when no TTL parameter is passed, for some reason.
```

Comment:
Seems this behavior is surprising to you.

Do you feel it should always be checked?

Reply:
I guess this may be okay now that we've changed the behavior for min/max password time. Here's the behavior as I understand it...

 - For native authentication, if no TTL is passed, the password never expires. We don't need to check the TTL in this case because there is no TTL.
 - For pam_password authentication, if no TTL is passed, the configured `min_password_time` is used. We don't need to check the TTL in this case because it is by definition in the range.

I think the "for some reason" is from the test I copied and it was confusing in the past because PAM/pam_password would use a third configuration as the default TTL if none was specified, and this value could fall outside the range of the min/max password times.

Reply:
Ok. Sounds like we can remove that part then. Your explanation seems to cover all cases.

You agree?

Reply:
Agreed. Added explanations in place of the "for some reason"s.